src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -270,14 +361,90 @@ - (void)test001_ReceiveOTAQuery
-    sOTAProviderDelegate.handleQueryImageExpectation = nil;
+    return responseExpectation;
+}
+
+- (void)test001_ReceiveOTAQuery
+{
+#if MANUAL_INDIVIDUAL_TEST
+    [self initStack];
+#endif
+
+    __auto_type * device = sConnectedDevice1;
+
+    XCTestExpectation * queryExpectation = [self expectationWithDescription:@"handleQueryImageForNodeID called"];
+    sOTAProviderDelegate.queryImageHandler = ^(NSNumber * nodeID, MTRDeviceController * controller,
+        MTROTASoftwareUpdateProviderClusterQueryImageParams * params, QueryImageCompletion completion) {
+        XCTAssertEqualObjects(nodeID, @(kDeviceId1));
+        XCTAssertEqual(controller, sController);
+        [sOTAProviderDelegate respondNotAvailableWithCompletion:completion];
+        [queryExpectation fulfill];
+    };
+
+    // Advertise ourselves as an OTA provider.
+    XCTestExpectation * announceResponseExpectation = [self announceProviderToDevice:device];
+
+    [self waitForExpectations:@[ queryExpectation, announceResponseExpectation ] timeout:kTimeoutInSeconds];
+
+    sOTAProviderDelegate.queryImageHandler = nil;
+}
+
+- (void)test002_ReceiveTwoQueriesExplicitBusy
+{
+#if MANUAL_INDIVIDUAL_TEST
+    [self initStack];
+#endif
+
+    // TODO: This test fails so far because the OTA requestor is following the
+    // spec and clamping the delay to a minimum of 120 seconds.  We don't really
+    // want to spend 2 minutes waiting in this test, so just disable it for
+    // now.  See https://github.com/project-chip/connectedhomeip/issues/25922
+    // for a proposal to address this.
+#if 0
+    __auto_type * device = sConnectedDevice1;
+
+    XCTestExpectation * queryExpectation1 = [self expectationWithDescription:@"handleQueryImageForNodeID called first time"];
+    XCTestExpectation * queryExpectation2 = [self expectationWithDescription:@"handleQueryImageForNodeID called second time"];
+    const uint16_t busyDelay = 1; // Second
+
+    __block QueryImageHandler handleSecondQuery;
+    sOTAProviderDelegate.queryImageHandler = ^(NSNumber * nodeID, MTRDeviceController * controller, MTROTASoftwareUpdateProviderClusterQueryImageParams * params,
+                                               QueryImageCompletion completion) {
+        sOTAProviderDelegate.queryImageHandler = handleSecondQuery;
+        XCTAssertEqualObjects(nodeID, @(kDeviceId1));
+        XCTAssertEqual(controller, sController);
+        [sOTAProviderDelegate respondBusyWithDelay:@(busyDelay) completion:completion];
+        [queryExpectation1 fulfill];
+    };
+
+    handleSecondQuery = ^(NSNumber * nodeID, MTRDeviceController * controller, MTROTASoftwareUpdateProviderClusterQueryImageParams * params,
+                                               QueryImageCompletion completion) {
+        XCTAssertEqualObjects(nodeID, @(kDeviceId1));
+        XCTAssertEqual(controller, sController);
+        [sOTAProviderDelegate respondNotAvailableWithCompletion:completion];
```

Comment:
are we trying to return available here or test both failure cases busy + not available? usually with busy the other request return available. Just trying to understand the test.

Reply:
For now this is just testing that if the delegate returns "busy" that causes the provider to later retry and call into the delegate again.

I'll adjust the comments explaining what this test is testing when I enable it, after https://github.com/project-chip/connectedhomeip/pull/25925 merges.